mitodl/odl-video-service
# Handle possible YT video dupes while checking statuses

#### What are the relevant tickets?
Fixes #486 

#### What's this PR do?
Catches `IndexError` when checking the status of a `YoutubeVideo` and sets the status to `YoutubeStatus.FAILED`.

#### How should this be manually tested?
- In the admin page for a video, set `is_public` to `True`
- in a django shell:
  ```python
  from cloudsync.tasks import upload_youtube_videos
  upload_youtube_videos.delay()
```
- Wait a few minutes or so for the video to upload to Youtube.
- Check the status of the `YoutubeVideo` object, hopefully it is `processed`
- Repeat the above steps, uploading the same video again.
- This time the status of the new ``YoutubeVideo` should be `failed`
- There should be an exception logged by celery.
